### ISSUE TITLE
Consider upstreaming changes to go-mastodon

### ISSUE THREAD

**Issue by mdosch** [RasmusLindroth/tut#276]
Dear developer,

thank you for this useful tool. I am considering packaging it for debian but I face a problem. For the debian packaging of golang programs it is necessary to have all dependencies also packaged. You use a fork of mattn/go-mastodon and write there `Do not use this version!`. So I would package mattn/go-mastodon instead of your fork but tut doesn't build with mattn/go-mastodon. So now I have to either package a library which clearly states it shall not be used (which feels wrong to me and maybe the debian ftp-masters would reject it due to this statement) or I'd have to vendor the go-mastodon library (which is strongly discouraged). Both possible solutions do not appeal to me and therefore I abstained from packaging this useful program.

Would you please consider one of the following solutions?

- Upstream your changes to mattn/go-mastodon
- Rework tut to be compatible to mattn/go-mastodon
- Commit yourself to maintain your fork of mattn/go-mastodon and remove the "Don't use this version!" warning

Best regards,
Martin

**Comment by r3k2** [comment]
yeah TUT is starting to fail to compile on new systems with newer versions of GO and such :(
